application/src/main/java/org/thingsboard/server/service/queue/ProtoUtils.java

```diff
@@ -62,4 +94,377 @@ public static EntityType fromProto(TransportProtos.EntityTypeProto entityType) {
+                .setEdgeIdMSB(request.getEdgeId().getId().getMostSignificantBits())
+                .setEdgeIdLSB(request.getEdgeId().getId().getLeastSignificantBits())
+                .build();
+    }
+
+    public static ToEdgeSyncRequest fromProto(TransportProtos.ToEdgeSyncRequestMsgProto proto) {
+        return new ToEdgeSyncRequest(
+                new UUID(proto.getRequestIdMSB(), proto.getRequestIdLSB()),
+                TenantId.fromUUID(new UUID(proto.getTenantIdMSB(), proto.getTenantIdLSB())),
+                new EdgeId(new UUID(proto.getEdgeIdMSB(), proto.getEdgeIdLSB()))
+        );
+    }
+
+    public static TransportProtos.FromEdgeSyncResponseMsgProto toProto(FromEdgeSyncResponse response) {
+        return TransportProtos.FromEdgeSyncResponseMsgProto.newBuilder()
+                .setTenantIdMSB(response.getTenantId().getId().getMostSignificantBits())
+                .setTenantIdLSB(response.getTenantId().getId().getLeastSignificantBits())
+                .setResponseIdMSB(response.getId().getMostSignificantBits())
+                .setResponseIdLSB(response.getId().getLeastSignificantBits())
+                .setEdgeIdMSB(response.getEdgeId().getId().getMostSignificantBits())
+                .setEdgeIdLSB(response.getEdgeId().getId().getLeastSignificantBits())
+                .setSuccess(response.isSuccess())
+                .build();
+    }
+
+    public static FromEdgeSyncResponse fromProto(TransportProtos.FromEdgeSyncResponseMsgProto proto) {
+        return new FromEdgeSyncResponse(
+                new UUID(proto.getResponseIdMSB(), proto.getResponseIdLSB()),
+                TenantId.fromUUID(new UUID(proto.getTenantIdMSB(), proto.getTenantIdLSB())),
+                new EdgeId(new UUID(proto.getEdgeIdMSB(), proto.getEdgeIdLSB())),
+                proto.getSuccess()
+        );
+    }
+
+    public static TransportProtos.EdgeEventUpdateMsgProto toProto(EdgeEventUpdateMsg msg) {
+        return TransportProtos.EdgeEventUpdateMsgProto.newBuilder()
+                .setTenantIdMSB(msg.getTenantId().getId().getMostSignificantBits())
+                .setTenantIdLSB(msg.getTenantId().getId().getLeastSignificantBits())
+                .setEdgeIdMSB(msg.getEdgeId().getId().getMostSignificantBits())
+                .setEdgeIdLSB(msg.getEdgeId().getId().getLeastSignificantBits())
+                .build();
+    }
+
+    public static EdgeEventUpdateMsg fromProto(TransportProtos.EdgeEventUpdateMsgProto proto) {
+        return new EdgeEventUpdateMsg(
+                TenantId.fromUUID(new UUID(proto.getTenantIdMSB(), proto.getTenantIdLSB())),
+                new EdgeId(new UUID(proto.getEdgeIdMSB(), proto.getEdgeIdLSB()))
+        );
+    }
+
+    private static TransportProtos.DeviceEdgeUpdateMsgProto toProto(DeviceEdgeUpdateMsg msg) {
+        return TransportProtos.DeviceEdgeUpdateMsgProto.newBuilder()
+                .setTenantIdMSB(msg.getTenantId().getId().getMostSignificantBits())
```

Comment:
This morning, such a spaghetti code produced some pain in my eyes. I believe a Java compiler that will optimize that stuff.

Reply:
Tbh, I don't get you... Do you suggest reformatting this code? A single line is not readable..

Reply:
Please feel free to leave it as is. We are using duplicated nested getters. It might be faster to get a local var in stack that costs nothing. And the lines will look much shorter. But my suggestion may cause a holly war between some communities so you can leave it as is. If you are keen to find the best approach, you can investigate a byte code produced by a Java compiler. You may find a situation in which the compiler already optimizes the code. Or find some hints on how to make an optimal code as well.
```
.setTenantIdMSB(tenantId.getMostSignificantBits())
```